ext/mbstring/mbstring.c

```diff
@@ -4587,13 +4590,183 @@ MBSTRING_API bool php_mb_check_encoding(const char *input, size_t length, const
 	return true;
 }
 
+static bool mb_fast_check_utf8(zend_string *str)
+{
+#ifdef __SSE2__
+	unsigned char *p = (unsigned char*)ZSTR_VAL(str), *e = p + ZSTR_LEN(str);
+	e -= sizeof(__m128i);
+
+	/* For checking for illegal bytes 0xF5-FF */
+	const __m128i over_f5 = _mm_set1_epi8(-117);
+	/* For checking for overlong 3-byte code units and reserved codepoints U+D800-DFFF */
+	const __m128i over_9f = _mm_set1_epi8(-97);
+	/* For checking for overlong 4-byte code units and invalid codepoints > U+10FFFF */
+	const __m128i over_8f = _mm_set1_epi8(-113);
+	/* For checking for illegal bytes 0xC0-C1 */
+	const __m128i find_c0 = _mm_set1_epi8(-64);
+	const __m128i c0_to_c1 = _mm_set1_epi8(-126);
+	/* For checking structure of continuation bytes */
+	const __m128i find_e0 = _mm_set1_epi8(-32);
+	const __m128i find_f0 = _mm_set1_epi8(-16);
+
+	__m128i last_block = _mm_setzero_si128();
+	__m128i operand;
+
+	while (p <= e) {
+		operand = _mm_loadu_si128((__m128i*)p); /* Load 16 bytes */
+
+check_operand:
+		/* If all 16 bytes are single-byte characters, there is no need to check anything */
+		__m128i multibyte = _mm_cmplt_epi8(operand, _mm_setzero_si128());
+		if (_mm_movemask_epi8(multibyte)) {
```

Comment:
This is syntactically invalid (label must not be followed by declaration). There is a simple trick, namely to introduce an empty statement (I think we use that already elsewhere):
```suggestion
check_operand:;
		/* If all 16 bytes are single-byte characters, there is no need to check anything */
		__m128i multibyte = _mm_cmplt_epi8(operand, _mm_setzero_si128());
		if (_mm_movemask_epi8(multibyte)) {
```

Reply:
Thanks! Fixed.